src/geofabrics/processor.py

```diff
@@ -481,5 +505,583 @@ def run(self):
             # interpolate
             self.dense_dem.interpolate_offshore(self.bathy_contours)
 
+        # Load in river bathymetry and incorporate where decernable at the resolution
+        if self.check_vector('river_polygons') and self.check_vector('river_bathymetry'):
+
+            # Get the polygons and bathymetry and can be multiple
+            bathy_dirs = self.get_vector_paths('river_bathymetry')
+            poly_dirs = self.get_vector_paths('river_polygons')
+
+            logging.info(f"Incorporating river Bathymetry: {bathy_dirs}")
+
+            # Load in bathymetry
+            self.river_bathy = geometry.RiverBathymetryPoints(
+                points_files=bathy_dirs,
+                polygon_files=poly_dirs,
+                catchment_geometry=self.catchment_geometry,
+                z_labels=self.get_instruction_general("river_bathy_z_label"))
+
+            # Call interpolate river on the DEM - note the DEM checks to see if any pixels actually fall inside the polygon
+            self.dense_dem.interpolate_river_bathymetry(river_bathymetry=self.river_bathy)
+
         # fill combined dem - save results
         self.dense_dem.dem.to_netcdf(self.get_instruction_path('result_dem'))
+
+
+class RiverBathymetryGenerator(BaseProcessor):
+    """ RiverbathymetryGenerator executes a pipeline to estimate river
+    bathymetry depths from flows, slopes, friction and widths along a main
+    channel. This is dones by first creating a hydrologically conditioned DEM
+    of the channel. A json_instructions file defines the pipeline logic and
+    data.
+
+    Attributes:
+        channel_polyline  The main channel along which to estimate depth. This
+            is a polyline.
+        gen_dem  The ground DEM generated along the main channel. This is a
+            raster.
+        veg_dem  The vegetation DEM generated along the main channel. This is a
+            raster.
+        aligned_channel_polyline  The main channel after its alignment has been
+            updated based on the DEM. Width and slope are estimated on this.
+        transects  Transect polylines perpindicular to the aligned channel with
+            samples of the DEM values
+    """
+
+    def __init__(self, json_instructions: json, debug: bool = True):
```

Comment:
Yes. I have some logging at different levels. The debug here is more to do with writing out a additional files that can be viewed elsewhere to get a better sense of the internal behaviour of the algorithm. Long term these may be removed in the future.

Reply:
@LukeParky. I'd like to keep this at least until the approach is published. At that point it can hopefully be removed or brought in as an option in a different way.